dandiapi/zarr/tests/utils.py

```diff
@@ -1,31 +1,18 @@
 from __future__ import annotations
 
-import hashlib
-import os
-from pathlib import Path
-from typing import TYPE_CHECKING
+import factory
 
-import faker
-from zarr_checksum.generators import ZarrArchiveFile
 
-from dandiapi.api.storage import get_boto_client
+class _PostgenerationAttributesFactoryOptions(factory.base.FactoryOptions):
+    def use_postgeneration_results(self, step, instance, results):
+        self.factory._after_postgeneration(
+            instance,
+            create=step.builder.strategy == factory.CREATE_STRATEGY,
+            results=results,
+            attributes=step.attributes,
+        )
```

Comment:
I don't really understand why this function is necessary, or where it's called. A comment explaining this might be helpful, since it seems like this is making use of undocumented behavior.

Reply:
Sure, I'll better document this class.